### ISSUE TITLE
python interface on mac os

### ISSUE THREAD

**Issue by germasch** [ornladios/ADIOS2#2774]
We ask the community to always report bugs. Please open a new issue if some other comes up in the conversation. It will help us stay on topic and manage tasks more efficiently.

**Describe the bug**

I've previously been able to install adios2 with its python interface on MacOS, but now I'm running into this problem:

```
 ⚙ kai@kais-mbp  ~/build/ADIOS2/build > master $ export PYTHONPATH=/Users/kai/build/ADIOS2/build/install/lib/python3.7/site-packages
 ⚙ kai@kais-mbp  ~/build/ADIOS2/build > master $ python
Python 3.7.10 (default, Jun  1 2021, 23:43:35)
[Clang 11.0.3 (clang-1103.0.32.62)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import adios2
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/kai/build/ADIOS2/build/install/lib/python3.7/site-packages/adios2/__init__.py", line 1, in <module>
    from .adios2 import *
ImportError: dlopen(/Users/kai/build/ADIOS2/build/install/lib/python3.7/site-packages/adios2/adios2.cpython-37m-darwin.so, 2): Library not loaded: @rpath/libadios2_cxx11.2.dylib
  Referenced from: /Users/kai/build/ADIOS2/build/install/lib/python3.7/site-packages/adios2/adios2.cpython-37m-darwin.so
  Reason: image not found
```

**To Reproduce**
This is using today's master. The python I'm using is installed by macports, I had to point cmake to it by setting `Pyton_EXECUTABLE`.

**Expected behavior**
A working adios2 python module

**Desktop (please complete the following information):**
 - OS/Platform: MacOS 10.15.7

**Additional context**
Interestingly, the rpath in the python module is set:
```
Load command 14
          cmd LC_RPATH
      cmdsize 48
         path /Users/kai/build/ADIOS2/build/lib (offset 12)
```

This doesn't seem quite right in the sense that it's the build location, not the install location (which would be `.../build/install/lib` in my case). However, the library still does exist in the build location (`/Users/kai/build/ADIOS2/build/lib/libadios2_cxx11.2.dylib`), so I would have thought it'd be found there, but clearly, it's not :(

**Comment by manauref** [comment]
@germasch did you by any chance find a solution to this?

**Comment by JasonRuonanWang** [comment]
This should have been fixed, as I did use the python binding a lot a few months back on Mac using both a conda Python and a Python I built myself. But I haven't run into any similar issues. 

I would high recommend conda environments when using the Python binding. That's the most stable environments I have used for ADIOS2. The Mac system Python, the home-brew Python and the Macports Python do have some weird issues when dealing with complicated dependent libraries especially in a Python and C++/C mixed workflow.
